src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -0,0 +1,259 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_NONMAXSUPPRESSION_HPP
+#define MIGRAPHX_GUARD_OPERATORS_NONMAXSUPPRESSION_HPP
+
+#include <cmath>
+#include <queue>
+#include <cstdint>
+#include <iterator>
+#include <migraphx/config.hpp>
+#include <migraphx/float_equal.hpp>
+#include <migraphx/algorithm.hpp>
+#include <migraphx/tensor_view.hpp>
+#include <migraphx/shape_for_each.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct nonmaxsuppression
+{
+    int center_point_box = 0;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.center_point_box, "center_point_box"));
+    }
+
+    std::string name() const { return "nonmaxsuppression"; }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        auto lens = inputs.front().lens();
+        std::vector<int64_t> out_lens(2);
+        out_lens.at(0) = lens.at(1);
+        out_lens.at(1) = 3;
+        return {shape::int64_type, out_lens};
+    }
+
+    struct box_info_ptr
+    {
+        float score{};
+        int64_t index{};
+
+        inline bool operator<(const box_info_ptr& rhs) const
+        {
+            return score < rhs.score || (float_equal(score, rhs.score) && index > rhs.index);
+        }
+    };
+
+    void max_min(float lhs, float rhs, float& min, float& max) const
+    {
+        if(lhs >= rhs)
+        {
+            min = rhs;
+            max = lhs;
+        }
+        else
+        {
+            min = lhs;
+            max = rhs;
+        }
+    }
+
+    inline bool suppress_by_iou(const float* boxes_data,
+                                int64_t box_index1,
+                                int64_t box_index2,
+                                float iou_threshold) const
+    {
+        float x1_min{};
+        float y1_min{};
+        float x1_max{};
+        float y1_max{};
+        float x2_min{};
+        float y2_min{};
+        float x2_max{};
+        float y2_max{};
+        float intersection_x_min{};
+        float intersection_x_max{};
+        float intersection_y_min{};
+        float intersection_y_max{};
+
+        const float* box1 = boxes_data + 4 * box_index1;
+        const float* box2 = boxes_data + 4 * box_index2;
+        // center_point_box_ only support 0 or 1
```

Comment:
should we be checking if the user passes in a value != 0 or 1? also comment has extra underscore

Reply:
Changed to a bool type, so no need  to do this check anymore.